iampolicymanagementv1/iam_policy_management_v1_examples_test.go

```diff
@@ -676,5 +681,400 @@ var _ = Describe(`IamPolicyManagementV1 Examples Tests`, func() {
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(policyTemplateCollection, "", "  ")
+			fmt.Println(string(b))
+
+			// end-list_policy_templates
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(policyTemplateCollection).ToNot(BeNil())
+		})
+		It(`CreatePolicyTemplate request example`, func() {
+			fmt.Println("\nCreatePolicyTemplate() result:")
+			// begin-create_policy_template
+
+			policyRole := &iampolicymanagementv1.Roles{
+				RoleID: core.StringPtr("crn:v1:bluemix:public:iam::::role:Viewer"),
+			}
+			v2PolicyGrant := &iampolicymanagementv1.Grant{
+				Roles: []iampolicymanagementv1.Roles{*policyRole},
+			}
+			v2PolicyControl := &iampolicymanagementv1.Control{
+				Grant: v2PolicyGrant,
+			}
+			serviceNameResourceAttribute := &iampolicymanagementv1.V2PolicyResourceAttribute{
+				Key:      core.StringPtr("serviceName"),
+				Operator: core.StringPtr("stringEquals"),
+				Value:    core.StringPtr("iam-access-management"),
+			}
+			policyResource := &iampolicymanagementv1.V2PolicyResource{
+				Attributes: []iampolicymanagementv1.V2PolicyResourceAttribute{
+					*serviceNameResourceAttribute},
+			}
+
+			templatePolicyModel := &iampolicymanagementv1.TemplatePolicy{
+				Type:        core.StringPtr("access"),
+				Description: core.StringPtr("Test Template"),
+				Resource:    policyResource,
+				Control:     v2PolicyControl,
+			}
+
+			createPolicyTemplateOptions := iamPolicyManagementService.NewCreatePolicyTemplateOptions(
+				examplePolicyTemplateName,
+				exampleAccountID,
+				templatePolicyModel,
+			)
+
+			policyTemplate, response, err := iamPolicyManagementService.CreatePolicyTemplate(createPolicyTemplateOptions)
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(policyTemplate, "", "  ")
+			fmt.Println(string(b))
+
+			// end-create_policy_template
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(201))
+			Expect(policyTemplate).ToNot(BeNil())
+
+			examplePolicyTemplateID = *policyTemplate.ID
+
+		})
+		It(`GetPolicyTemplate request example`, func() {
+			fmt.Println("\nGetPolicyTemplate() result:")
+			// begin-get_policy_template
+
+			getPolicyTemplateOptions := iamPolicyManagementService.NewGetPolicyTemplateOptions(
+				examplePolicyTemplateID,
+			)
+
+			policyTemplate, response, err := iamPolicyManagementService.GetPolicyTemplate(getPolicyTemplateOptions)
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(policyTemplate, "", "  ")
+			fmt.Println(string(b))
+
+			// end-get_policy_template
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(policyTemplate.AccountID).ToNot(BeNil())
+			Expect(policyTemplate.Version).ToNot(BeNil())
+			Expect(policyTemplate.Name).ToNot(BeNil())
+			Expect(policyTemplate.Policy).ToNot(BeNil())
+
+			examplePolicyTemplateETag = response.GetHeaders().Get("ETag")
```

Comment:
You can also use a link to store off ETag values and then use the value to initialize an "IfMatch" header param in downstream operations.